homeassistant/components/ibeacon/coordinator.py

```diff
@@ -296,12 +307,24 @@ def _async_update_ibeacon_with_unique_address(
         address = service_info.address
         unique_id = f"{group_id}_{address}"
         new = unique_id not in self._last_ibeacon_advertisement_by_unique_id
-        # Reject creating new trackers if the name is not set
-        if new and (
-            service_info.device.name is None
-            or service_info.device.name.replace("-", ":") == service_info.device.address
+        uuid = str(ibeacon_advertisement.uuid)
+
+        # Reject creating new trackers if the name is not set (unless the uuid is allowlisted).
+        if (
+            new
+            and uuid not in self._entry.options.get(CONF_ALLOW_NAMELESS_UUIDS, [])
```

Comment:
Save `self._entry.options.get(CONF_ALLOW_NAMELESS_UUIDS, [])` in a `set` in the object to avoid having to do a linear search over it each time

Reply:
Done.